server/test/unit/server/webservices/views/test_task_groups.py

```diff
@@ -27,6 +28,55 @@ def filter(self, state=None):
         return MockQuerySet(filtered_list)
 
 
+class TestTaskGroupView(unittest.TestCase):
+    """
+    Tests for TaskGroupView
+    """
+    @mock.patch('pulp.server.webservices.views.decorators._verify_auth',
+                new=assert_auth_DELETE())
+    @mock.patch('pulp.server.webservices.views.task_groups.TaskStatus.objects')
+    @mock.patch('pulp.server.webservices.views.task_groups.tasks')
+    def test_delete_task_resource_nonexistant(self, mock_task, mock_objects):
+        """
+        Test delete task_group with no tasks
+        """
+        mock_request = mock.MagicMock()
+        mock_objects.only.return_value.get_or_404.side_effect = MissingResource
+        task_resource = TaskGroupView()
+        try:
+            response = task_resource.delete(mock_request, 'mock_task')
+        except MissingResource, response:
+            pass
+        else:
+            AssertionError("MissingResource should be raised with non-existing task group.")
+
+        self.assertEqual(response.http_status_code, 404)
+
+    @mock.patch('pulp.server.webservices.views.decorators._verify_auth',
+                new=assert_auth_DELETE())
+    @mock.patch('pulp.server.webservices.views.task_groups.TaskStatus.objects')
+    @mock.patch('pulp.server.webservices.views.task_groups.tasks')
+    @mock.patch(
+        'pulp.server.webservices.views.task_groups.generate_json_response')
+    def test_delete_task_resource(self, mock_resp, mock_task, mock_objects):
+        """
+        Test delete task_group with one task
+        """
+        mock_request = mock.MagicMock()
+        task1 = mock.Mock(task_id='test_foo_path')
+        task2 = mock.Mock(task_id='test_foo_path1')
+        mock_objects.only.return_value.get_or_404.return_value = [task1, task2]
+
+        task_resource = TaskGroupView()
+
+        response = task_resource.delete(mock_request, 'mock_task')
+
+        assert 2 == mock_task.cancel.call_count
```

Comment:
use ``self.assertEqual`` instead to keep this in the realm of the test framework.